src/common/Makefile.am

```diff
@@ -86,11 +87,6 @@ libcommon_internal_la_SOURCES += \
 	common/pick_address.cc
 endif
 
-if LINUX
```

Comment:
@tchaikov  I don't have a Solaris to check, but linux_versoin.c uses uname(). It looks like it is present on Solaris too and compatible.